lighthouse-cli/test/smokehouse/lighthouse-runners/devtools.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview A runner that launches Chrome and executes Lighthouse via DevTools.
+ */
+
+import fs from 'fs';
+import os from 'os';
+import util from 'util';
+import {execFile} from 'child_process';
+
+const execFilePromise = util.promisify(execFile);
+
+/** @type {Promise<void>} */
+let setupPromise;
+async function setup() {
+  if (setupPromise) return setupPromise;
+
+  setupPromise = new Promise(resolve => {
+    // execFileSync('yarn', ['open-devtools']);
```

Comment:
TODO: I thought about doing the setup for building ToT devtools here... would that be useful?

Reply:
> TODO: I thought about doing the setup for building ToT devtools here... would that be useful?

yeah, though it would be nice if it didn't attempt to re-update on every run. Is it currently only the content-shell we only update every X commits?

Reply:
The GHA workflow already builds devtools before starting this smoke test as part of the layout tests ... so let's do something here but only for local runs.

```
export DEVTOOLS_PATH=".tmp/devtools/devtools-frontend"
node lighthouse-core/test/chromium-web-tests/download-devtools.sh
node lighthouse-core/test/chromium-web-tests/roll-devtools.sh
```

(and then the local devtools smoke run would defer to w/e CHROME_PATH you have, not use the latest content shell ... unless thats your chrome path.)